File: composite/c-sharp/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Composite
{
    // Somut yapımızdır.
    // UML diyagramındaki Leaf kısmına denk gelmektedir.
    // Hiyerarşinin en alt tabakasını temsil etmektedir.
    class Product : ICatalogComponent
    {
        private string _name;

        public Product(string name)
        {
            _name = name;
        }

        public void DrawHierarchy()
        {
            Console.WriteLine(_name);
        }
    }
}

File: composite/python/product.py
from catalog_component import CatalogComponent


class Product(CatalogComponent):
    """
    Somut yapımızdır.
    UML diyagramındaki Leaf kısmına denk gelmektedir.
    Hiyerarşinin en alt tabakasını temsil etmektedir.
    """

    def __init__(self, name):
        self._name = name

    def draw_hierarchy(self):
        print(self._name)
